x-pack/legacy/plugins/ml/server/new_platform/plugin.ts

```diff
@@ -69,6 +67,7 @@ export interface MlCoreSetup {
   injectUiAppVars: (id: string, callback: () => {}) => any;
   http: MlHttpServiceSetup;
   savedObjects: SavedObjectsLegacyService;
+  coreSavedObjects: SavedObjectsServiceStart;
```

Comment:
You might save yourself some time by naming this the same as it is on CoreStart:

```suggestion
  savedObjects: SavedObjectsServiceStart;
```

Reply:
I see that you're using that name already for the legacy interface. I recommend switching that to a separate "legacy" object that gets passed into `start`, similar to the second code example in this section (look for `LegacySetup`): https://github.com/elastic/kibana/blob/master/src/core/MIGRATION.md#introduce-new-plugin-definition-shim

Reply:
Thanks, @joshdover. 😄 I think at this point, we're close enough to just cut over fully to NP so instead of making this change now I'll go ahead and switch over to NP in a follow-up PR